subsys/usb/class/cdc_acm.c

```diff
@@ -1074,30 +1074,33 @@ static const struct uart_driver_api cdc_acm_driver_api = {
 }
 #endif /* CONFIG_USB_COMPOSITE_DEVICE */
 
-#define DEFINE_CDC_ACM_DEV_DATA(x)					\
+#define DEFINE_CDC_ACM_DEV_DATA(x, _)					\
 	static struct cdc_acm_dev_data_t cdc_acm_dev_data_##x = {	\
 		.usb_status = USB_DC_UNKNOWN,				\
 		.line_coding = CDC_ACM_DEFAUL_BAUDRATE,			\
-}
+	};
 
-#define DEFINE_CDC_ACM_DEVICE(x)					\
-	DEVICE_AND_API_INIT(cdc_acm_##x, CONFIG_CDC_ACM_PORT_NAME_##x,	\
+#define DEFINE_CDC_ACM_DEVICE(x, _)					\
+	DEVICE_AND_API_INIT(cdc_acm_##x,				\
+			    CONFIG_USB_CDC_ACM_DEVICE_NAME "_" #x,	\
 			    &cdc_acm_init, &cdc_acm_dev_data_##x,	\
 			    &cdc_acm_config_##x,			\
 			    APPLICATION,				\
 			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		\
-			    &cdc_acm_driver_api)
-
-DEFINE_CDC_ACM_DESCR(0, 0x80, 0x00, 0x80);
-DEFINE_CDC_ACM_EP(0, 0x80, 0x00, 0x80);
-DEFINE_CDC_ACM_CFG_DATA(0);
-DEFINE_CDC_ACM_DEV_DATA(0);
-DEFINE_CDC_ACM_DEVICE(0);
-
-#if defined(CONFIG_CDC_ACM_PORT_NAME_1)
-DEFINE_CDC_ACM_DESCR(1, 0x80, 0x00, 0x80);
-DEFINE_CDC_ACM_EP(1, 0x80, 0x00, 0x80);
-DEFINE_CDC_ACM_CFG_DATA(1);
-DEFINE_CDC_ACM_DEV_DATA(1);
-DEFINE_CDC_ACM_DEVICE(1);
-#endif
+			    &cdc_acm_driver_api);
+
+/* Use Automatic endpoint assignment */
+#define AUTO_EP_IN	0x80
```

Comment:
@jfischer-phytec-iot OK, what about assigning those default values to AUTO_EP_IN ?

Reply:
Up to you, but do not assign the same endpoint address for more than one endpoint, you can rename the macros above or just use it:

```c
#define DEFINE_CDC_ACM_DESCR_AUTO(x, _) \
	DEFINE_CDC_ACM_DESCR(x, CDC_ACM_INT_EP_ADDR, CDC_ACM_OUT_EP_ADDR, CDC_ACM_IN_EP_ADDR);

#define DEFINE_CDC_ACM_EP_AUTO(x, _) \
	DEFINE_CDC_ACM_EP(x, CDC_ACM_INT_EP_ADDR, CDC_ACM_OUT_EP_ADDR, CDC_ACM_IN_EP_ADDR);
```

Reply:
@finikorg You do not need AUTO_EP_IN and AUTO_EP_OUT macros, the default addresses are already [defined,](https://github.com/zephyrproject-rtos/zephyr/pull/13310/commits/a23cb10f29f4a499c88fe76c093b73b11a2fa0a5#diff-e33cccf141c95f57405ce187a047a007R76) see my comment above.

Reply:
@jfischer-phytec-iot I forgot to delete these definition. Do you want to keep those definitions?

Reply:
@finikorg I would like to keep them, but you can change default values if you want.

Reply:
@jfischer-phytec-iot I think we should use auto endpoint configuration especially those default values are not correct for all platform ( for example https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/usb/device/usb_dc_kinetis.c#L284)

Reply:
@finikorg The default value of a endpoint address  will be reassigned, it just have to be valid endpoint address and unique inside the interface. In the quoted kinetis driver, the half of the endpoints is just pre-selected as IN.

Reply:
@jfischer-phytec-iot I have test where you can see that proposed way works just fine, could you check this last commit 46b6e56b193f376f1d5f6f08c0a365a57af272e8.
You can run it in native_posix and it will print endpoint assigned.